contracts/token/erc721/abstract/ImmutableERC721Base.sol

```diff
@@ -34,8 +32,24 @@ abstract contract ImmutableERC721Base is
     /// @dev Common URIs for individual token URIs
     string public baseURI;
 
-    /// @dev the tokenId of the next NFT to be minted.
-    Counters.Counter private nextTokenId;
+    /// @dev Only MINTER_ROLE can invoke permissioned mint.
+    bytes32 public constant MINTER_ROLE = bytes32("MINTER_ROLE");
+
+    /// @dev Total amount of minted tokens to a non zero address
+    uint256 public _totalSupply = 0;
```

Comment:
nit; you don't have to initalise to 0 here, as that is the default value. `uint256 public _totalSupply;`